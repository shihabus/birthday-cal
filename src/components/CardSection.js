import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Card from "./Card";

const Wrapper = styled.section`
  display: flex;
  overflow: auto;
  grid-area: list;
  width: 100%;

  justify-content: flex-start;
  align-content: flex-start;
`;

const List = styled.div`
  display: flex;
  overflow: auto;
  width: 100%;

  justify-content: flex-start;
  align-items: center;
  align-content: flex-start;
`;

export default function CardSection(props) {
  const { data } = props;
  const keys = Object.keys(data);
  return (
    <Wrapper>
      {keys.length === 0 ? (
        <Card isEmpty />
      ) : (
        <List>
          {keys.map(item => (
            <Card key={item} data={data[item]} />
          ))}
        </List>
      )}
    </Wrapper>
  );
}

CardSection.defaultProps = {
  data: {}
};

CardSection.propTypes = {
  data: PropTypes.objectOf(PropTypes.array)
};
