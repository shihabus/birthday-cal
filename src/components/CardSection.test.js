import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import CardSection from "./CardSection";

test("CardSection snapshot", () => {
  const { container } = render(<CardSection />);
  expect(container.firstChild).toMatchSnapshot();
});
