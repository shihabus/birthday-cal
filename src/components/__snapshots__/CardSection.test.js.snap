// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSection snapshot 1`] = `
<section
  class="sc-AxhUy iwoEhU"
>
  <div
    class="sc-AxirZ hGmzvG"
  >
    <div
      class="sc-AxiKw jxIpNn"
    >
      Sun
    </div>
    <section
      class="sc-AxhCb gWzavo"
    >
      <div
        class="sc-AxjAm pncij"
      >
        <span
          aria-label="sad emoji"
          role="img"
          style="font-size: 4rem;"
        >
          ðŸ˜”
        </span>
      </div>
    </section>
  </div>
</section>
`;
